providers/fips/fipsprov.c

```diff
@@ -553,19 +721,18 @@ int OSSL_provider_init(const OSSL_PROVIDER *provider,
         return 0;
     }
 
-    *out = fips_dispatch_table;
-    *provctx = ctx;
-
     /*
      * TODO(3.0): Remove me. This is just a dummy call to demonstrate making
      * EVP calls from within the FIPS module.
      */
-    if (!dummy_evp_call(*provctx)) {
-        OPENSSL_CTX_free(*provctx);
-        *provctx = NULL;
+    if (!dummy_evp_call(ctx)) {
```

Comment:
I have moved this call up here because it actually hangs when exiting if there was an error. I didnt try to track down the real issue as this is temporary call that will be removed at some point.